tests/src/JIT/Performance/CodeQuality/HWIntrinsic/X86/PacketTracer/PacketTracer.cs

```diff
@@ -0,0 +1,260 @@
+            ColorPacket256 lcolor = new ColorPacket256(lcolorR, lcolorG, lcolorB);
+
+            Vector256<float> specular = VectorPacket256.DotProduct(livec, rds.Normalize());
+            Vector256<float> specularGraterThanZero = Compare(specular, zero, FloatComparisonMode.GreaterThanOrderedNonSignaling);
+
+            var difColor = new ColorPacket256(1, 1, 1);
+            var splColor = new ColorPacket256(1, 1, 1);
+            var roughness = SetAllVector256<float>(1);
+
+            for (int j = 0; j < scene.Things.Length; j++)
+            {
+                Vector256<float> thingMask = StaticCast<int, float>(CompareEqual(things, SetAllVector256<int>(j)));
+                var rgh = SetAllVector256<float>(scene.Things[j].Surface.Roughness);
+                var dif = scene.Things[j].Surface.Diffuse(pos);
+                var spl = scene.Things[j].Surface.Specular;
+
+                roughness = BlendVariable(roughness, rgh, thingMask);
+
+                difColor.Xs = BlendVariable(difColor.Xs, dif.Xs, thingMask);
+                difColor.Ys = BlendVariable(difColor.Ys, dif.Ys, thingMask);
+                difColor.Zs = BlendVariable(difColor.Zs, dif.Zs, thingMask);
+
+                splColor.Xs = BlendVariable(splColor.Xs, spl.Xs, thingMask);
+                splColor.Ys = BlendVariable(splColor.Ys, spl.Ys, thingMask);
+                splColor.Zs = BlendVariable(splColor.Zs, spl.Zs, thingMask);
+            }
+
+            var tmpColor2 = VectorMath.Pow(specular, roughness) * colorPacket;
+            Vector256<float> scolorR = BlendVariable(defaultRGB, tmpColor2.Xs, specularGraterThanZero);
+            Vector256<float> scolorG = BlendVariable(defaultRGB, tmpColor2.Ys, specularGraterThanZero);
+            Vector256<float> scolorB = BlendVariable(defaultRGB, tmpColor2.Zs, specularGraterThanZero);
+            ColorPacket256 scolor = new ColorPacket256(scolorR, scolorG, scolorB);
+
+            var oldColor = colors;
+
+            colors = colors + ColorPacket256Helper.Times(difColor, lcolor) + ColorPacket256Helper.Times(splColor, scolor);
+
+            colors = new ColorPacket256(BlendVariable(colors.Xs, oldColor.Xs, isInShadow), BlendVariable(colors.Ys, oldColor.Ys, isInShadow), BlendVariable(colors.Zs, oldColor.Zs, isInShadow));
+
+        }
+        return colors;
+    }
+
+    private ColorPacket256 GetReflectionColor(Vector256<int> things, VectorPacket256 pos, VectorPacket256 norms, VectorPacket256 rds, Scene scene, int depth)
+    {
+        return scene.Reflect(things, pos) * TraceRay(new RayPacket256(pos, rds), scene, depth + 1);
+    }
+
+    private VectorPacket256 GetPoints(Vector256<float> x, Vector256<float> y, Camera camera)
+    {
+        float widthRate1 = Width / 2.0f;
+        float widthRate2 = Width * 2.0f;
+
+        float heightRate1 = Height / 2.0f;
+        float heightRate2 = Height * 2.0f;
+
+        var recenteredX = Divide(Subtract(x, SetAllVector256(widthRate1)), SetAllVector256(widthRate2));
```

Comment:
Is this more or less perforamnt than a `SetAllVector256(Width)` and `SetAllVector256(2)` along with a `Multiple` and `Divide` call?